src/parser/csql_grammar.y

```diff
@@ -3188,6 +3185,9 @@ class_name_for_synonym
 		    }
 		  cname->info.name.original = pt_append_string (this_parser, cname->info.name.original, "@");
 		  cname->info.name.original = pt_append_string (this_parser, cname->info.name.original, sname->info.name.original);
+
+		  PT_NAME_INFO_SET_FLAG (cname, PT_NAME_INFO_SERVER_SPECIFIED);
```

Comment:
Most of the rules check if a node is NULL.
How about checking if cname and sname are NULL for consistency?
```
if (cname && sname)
  {
    ...
  }
```

Reply:
It was included in commit e23317e.